File: src/KissU.Service/Queries/Systems/UserQuery.cs
using System;
using System.ComponentModel.DataAnnotations;
using Util;
using Util.Datas.Queries;

namespace KissU.Service.Queries.Systems {
    /// <summary>
    /// 用户查询参数
    /// </summary>
    public class UserQuery : QueryParameter {
        /// <summary>
        /// 
        /// </summary>
        [Display(Name="")]
        public Guid? UserId { get; set; }
        
        private string _userName = string.Empty;
        /// <summary>
        /// 用户名
        /// </summary>
        [Display(Name="用户名")]
        public string UserName {
            get => _userName == null ? string.Empty : _userName.Trim();
            set => _userName = value;
        }
        
        private string _realName = string.Empty;
        /// <summary>
        /// 真实姓名
        /// </summary>
        [Display(Name="真实姓名")]
        public string RealName {
            get => _realName == null ? string.Empty : _realName.Trim();
            set => _realName = value;
        }
        
        private string _nickName = string.Empty;
        /// <summary>
        /// 昵称
        /// </summary>
        [Display(Name="昵称")]
        public string NickName {
            get => _nickName == null ? string.Empty : _nickName.Trim();
            set => _nickName = value;
        }
        
        private string _email = string.Empty;
        /// <summary>
        /// 邮箱
        /// </summary>
        [Display(Name="邮箱")]
        public string Email {
            get => _email == null ? string.Empty : _email.Trim();
            set => _email = value;
        }
        /// <summary>
        /// 邮箱是否验证
        /// </summary>
        [Display(Name="邮箱是否验证")]
        public bool? EmailConfirmed { get; set; }
        
        private string _phoneNumber = string.Empty;
        /// <summary>
        /// 手机号码
        /// </summary>
        [Display(Name="手机号码")]
        public string PhoneNumber {
            get => _phoneNumber == null ? string.Empty : _phoneNumber.Trim();
            set => _phoneNumber = value;
        }
        /// <summary>
        /// 手机号码是否验证
        /// </summary>
        [Display(Name="手机号码是否验证")]
        public bool? PhoneNumberConfirmed { get; set; }
        
        private string _password = string.Empty;
        /// <summary>
        /// 密码
        /// </summary>
        [Display(Name="密码")]
        public string Password {
            get => _password == null ? string.Empty : _password.Trim();
            set => _password = value;
        }
        
        private string _passwordHash = string.Empty;
        /// <summary>
        /// 密码哈希值
        /// </summary>
        [Display(Name="密码哈希值")]
        public string PasswordHash {
            get => _passwordHash == null ? string.Empty : _passwordHash.Trim();
            set => _passwordHash = value;
        }
        /// <summary>
        /// 起始
        /// </summary>
        [Display( Name = "起始" )]
        public DateTime? BeginCreationTime { get; set; }
        /// <summary>
        /// 结束
        /// </summary>
        [Display( Name = "结束" )]
        public DateTime? EndCreationTime { get; set; }
        /// <summary>
        /// 
        /// </summary>
        [Display(Name="")]
        public Guid? CreatorId { get; set; }
        /// <summary>
        /// 起始
        /// </summary>
        [Display( Name = "起始" )]
        public DateTime? BeginLastModificationTime { get; set; }
        /// <summary>
        /// 结束
        /// </summary>
        [Display( Name = "结束" )]
        public DateTime? EndLastModificationTime { get; set; }
        /// <summary>
        /// 
        /// </summary>
        [Display(Name="")]
        public Guid? LastModifierId { get; set; }
    }
}

File: template/CodeSmith/CodeGenerator/02-Result/Presentation/Typings/app/systems/user/model/user-query.ts
import { QueryParameter } from '../../../../util';

/**
 * 用户查询参数
 */
export class UserQuery extends QueryParameter {
    /**
     * 用户标识
     */
    userId;
    /**
     * 用户名
     */
    userName;
    /**
     * 标准化用户名
     */
    normalizedUserName;
    /**
     * 安全邮箱
     */
    email;
    /**
     * 标准化邮箱
     */
    normalizedEmail;
    /**
     * 邮箱已确认
     */
    emailConfirmed;
    /**
     * 安全手机
     */
    phoneNumber;
    /**
     * 手机已确认
     */
    phoneNumberConfirmed;
    /**
     * 密码
     */
    password;
    /**
     * 密码散列
     */
    passwordHash;
    /**
     * 安全码
     */
    safePassword;
    /**
     * 安全码散列
     */
    safePasswordHash;
    /**
     * 启用两阶段认证
     */
    twoFactorEnabled;
    /**
     * 启用
     */
    enabled;
    /**
     * 起始冻结时间
     */
    beginDisabledTime;
    /**
     * 结束冻结时间
     */
    endDisabledTime;
    /**
     * 启用锁定
     */
    lockoutEnabled;
    /**
     * 锁定截止
     */
    lockoutEnd;
    /**
     * 登录失败次数
     */
    accessFailedCount;
    /**
     * 登录次数
     */
    loginCount;
    /**
     * 注册Ip
     */
    registerIp;
    /**
     * 起始上次登陆时间
     */
    beginLastLoginTime;
    /**
     * 结束上次登陆时间
     */
    endLastLoginTime;
    /**
     * 上次登陆Ip
     */
    lastLoginIp;
    /**
     * 起始本次登陆时间
     */
    beginCurrentLoginTime;
    /**
     * 结束本次登陆时间
     */
    endCurrentLoginTime;
    /**
     * 本次登陆Ip
     */
    currentLoginIp;
    /**
     * 安全戳
     */
    securityStamp;
    /**
     * 备注
     */
    remark;
    /**
     * 起始创建时间
     */
    beginCreationTime;
    /**
     * 结束创建时间
     */
    endCreationTime;
    /**
     * 创建人
     */
    creatorId;
    /**
     * 起始最后修改时间
     */
    beginLastModificationTime;
    /**
     * 结束最后修改时间
     */
    endLastModificationTime;
    /**
     * 最后修改人
     */
    lastModifierId;
}